lib/trento/infrastructure/software_updates/suma.ex

```diff
@@ -86,24 +87,27 @@ defmodule Trento.Infrastructure.SoftwareUpdates.Suma do
 
   defp do_handle({:get_system_id, fully_qualified_domain_name}, %State{
          url: url,
-         auth: auth_cookie
+         auth: auth_cookie,
+         ca_cert: ca_cert
        }),
-       do: SumaApi.get_system_id(url, auth_cookie, fully_qualified_domain_name)
+       do: SumaApi.get_system_id(url, auth_cookie, fully_qualified_domain_name, ca_cert)
 
   defp do_handle({:get_relevant_patches, system_id}, %State{
          url: url,
-         auth: auth_cookie
+         auth: auth_cookie,
+         ca_cert: ca_cert
        }),
-       do: SumaApi.get_relevant_patches(url, auth_cookie, system_id)
+       do: SumaApi.get_relevant_patches(url, auth_cookie, system_id, ca_cert)
 
-  defp process_identifier(server_name), do: {:global, identificaton_tuple(server_name)}
+  defp process_identifier(server_name), do: {:global, identification_tuple(server_name)}
 
-  defp identificaton_tuple(server_name), do: {__MODULE__, server_name}
+  defp identification_tuple(server_name), do: {__MODULE__, server_name}
 
   defp setup_auth(%State{} = state) do
     with {:ok, %{url: url, username: username, password: password, ca_cert: ca_cert}} <-
            SoftwareUpdates.get_settings(),
-         {:ok, auth_cookie} <- SumaApi.login(url, username, password) do
+         :ok <- maybe_write_ca_cert_file(ca_cert),
+         {:ok, auth_cookie} <- SumaApi.login(url, username, password, ca_cert) do
```

Comment:
If we decide to have `SumaApi` functions accept a `use_ca_cert` as mentioned [here](https://github.com/trento-project/web/pull/2391/files#r1515701175) we could also think of centralizing that `ca_cert =! nil` in the genserver state that we set in this function :eyes: